### DISCUSSION TITLE
Timeline feature similar to Tawk.to

### DISCUSSION THREAD

**Discussion by ghost** [chatwoot/chatwoot#4754] [Feature Requests]
Hello Chatwoot community,

I wanted to put forward my suggestion for a Timeline feature that I find very useful in one of Chatwoots competitors, Tawk.to. 

Timeline allows you to see a visitor's entire navigation history from the agent chat window. From the referring URL to their navigation between pages on your site. It's shown in chronological order from top to bottom with timestamps. 

I have found this feature to be helpful in these areas: 

- Gives context when providing support to visitors (e.g., have they visited the FAQ? Did they use the Support section of the site to look up their issue?)
- It provides feedback to Support agents when users click links. The timeline is updated instantly when a user clicks a link, you can see they went to the URL you provided.
- Helped find many 404 issues with bad links / missing pages. 
- Give context for Sales chats (e.g., customer was looking into X & Y features, I should restate the benefits of those features while chatting with them)
- Feedback for site design / flow. It's helped me understand how customers explore my site by seeing common patterns by studying URL history of visitors from the chat dashboard. (e.g., Started at X, went to Y, ended up at Z. )

Below is an example of the Timeline in tawk.to:
![timeline](https://user-images.githubusercontent.com/29169428/170372710-a6d18ea2-888e-4f59-a277-95f7965aedb4.png)
 
I'm sure there are other area where this feature would be useful and not just my particular use case.

**Comment by sojan-official** [comment]
We have it in our backlog to add an events module to Chatwoot CRM
Which should help in building a customer timeline 

ref: https://github.com/chatwoot/chatwoot/issues/2271
